addons/payment_authorize/models/payment_transaction.py

```diff
@@ -0,0 +1,128 @@
+
+from odoo import _, models
+
+from odoo.addons.payment import utils as payment_utils
+from odoo.exceptions import UserError
+
+from .authorize_request import AuthorizeAPI
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    def _get_specific_processing_values(self, processing_values):
+        if self.acquirer_id.provider != 'authorize':
+            return super()._get_specific_processing_values(processing_values)
+
+        return {
+            'access_token': payment_utils.generate_access_token(
+                self.env['ir.config_parameter'].sudo().get_param('database.secret'),
+                processing_values['reference'],
+                processing_values['partner_id']
+            )
+        }
+
+    def _process_feedback_data(self, data):
+        if self.provider != 'authorize':
+            return super()._process_feedback_data(data)
+
+        if not self.acquirer_reference:
+            self.acquirer_reference = data.get('x_trans_id')
+
+        status_code = int(data.get('x_response_code', '3'))
+        if status_code == 1:  # Valid
+            status_type = data.get('x_type').lower()
+            if status_type in ['auth_capture', 'prior_auth_capture']:
+                self._set_done()
+            elif status_type == 'auth_only':
+                self._set_authorized()
+            elif status_type == 'void':
+                self._set_canceled()
+
+            if self.tokenize and not self.token_id:
+                authorize = AuthorizeAPI(self.acquirer_id)
+                cust_profile = authorize.create_customer_profile_from_tx(self.partner_id, self.acquirer_reference)
+                if cust_profile:
+                    token = self.env['payment.token'].create({
+                        'authorize_profile': cust_profile.get('profile_id'),
+                        'name': cust_profile.get('name'),
+                        'acquirer_ref': cust_profile.get('payment_profile_id'),
+                        'acquirer_id': self.acquirer_id.id,
+                        'partner_id': self.partner_id.id,
+                    })
+                    self.token_id = token
+        elif status_code == 2:  # Declined
+            self._set_canceled()
+        elif status_code == 4:  # Pending
+            self._set_pending()
+        else:
+            error = data.get('x_response_reason_text')
+            _logger.info("Authorize: " + _('error received whilst processing payment:\n%s', error))
+            self._set_error(error)
+
+    def _authorize_make_payment(self, opaque_data):
+        self.ensure_one()
+        authorize = AuthorizeAPI(self.acquirer_id)
+        if self.acquirer_id.capture_manually or self.operation == 'validation':
+            res = authorize.authorize(self.amount, self.reference, opaque_data=opaque_data)
+        else:
+            res = authorize.auth_and_capture(self.amount, self.reference, opaque_data=opaque_data)
```

Comment:
Why isn't the amount rounded here ? (it is rounded below when going through the "same" authorize route)

Reply:
Seems to be an oversight, it should be rounded indeed